source/solvers/navier_stokes_base.cc

```diff
@@ -871,6 +869,33 @@ NavierStokesBase<dim, VectorType, DofsType>::postprocess(bool firstIter)
         }
     }
 
+  if (this->nsparam.post_processing.calculate_average_velocities)
+    {
+      double epsilon = 1e-6;
+      // Reinitiation of the average_solution vector at the first iteration.
+      if (firstIter)
+        {
+          this->average_solution.reinit(this->locally_owned_dofs,
+                                        this->locally_relevant_dofs,
+                                        this->mpi_communicator);
```

Comment:
This could be re-inited in setup_dofs, but it is not a bad idea to have put it there also.